libs/platform/ebpf_program_types.c

```diff
@@ -65,23 +70,24 @@ ebpf_program_information_decode(
     }
 
     *program_information = local_program_information;
+    return_value = EBPF_SUCCESS;
 
 Done:
     if (handle)
         MesHandleFree(handle);
     ebpf_free(local_buffer);
 
-    return EBPF_SUCCESS;
+    return return_value;
 }
 
-void* __RPC_USER
-MIDL_user_allocate(size_t size)
+_Must_inspect_result_ _Ret_maybenull_ _Post_writable_byte_size_(size) void* __RPC_USER
+    MIDL_user_allocate(_In_ size_t size)
```

Comment:
```suggestion
    MIDL_user_allocate(size_t size)
```
don't use SAL annotation on value arguments, only reference arguments